Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs

```diff
@@ -507,7 +545,9 @@ eval BackPressed state = do
     continue state
 
 eval TriggerMaps state = continueWithCmd state[ do
-  _ <- pure $ openNavigation state.data.activeRide.dest_lat state.data.activeRide.dest_lon "DRIVE"
+  let destLat = if (state.data.activeRide.tripType == ST.Rental) then fromMaybe state.data.activeRide.src_lat state.data.activeRide.nextStopLat else state.data.activeRide.dest_lat
```

Comment:
in case it's rental ride and nextStopLat is Nothing, we should not use src_lat of the active ride and start the navigation. This should be handled properly